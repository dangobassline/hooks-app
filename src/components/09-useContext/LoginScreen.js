import React, {useContext} from 'react';
import { UserContext } from './UserContext';

export const LoginScreen = () => {

    // 1. Obtener la referencia al context userContext
    // 2. setUser
    // 3.

    const {setUser} = useContext(UserContext);

    return (
        <div>
            <h1>LoginScreen</h1>
            <hr/>
            <button
                className="btn btn-primary"
                onClick={ () => setUser({
                    id: 123,
                    name: "Dango"
                }) }
            >
            Login
            </button>
        </div>
    )
}
