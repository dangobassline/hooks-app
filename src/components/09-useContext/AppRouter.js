import React from 'react';
import {
  BrowserRouter as Router,
  Switch,
  Route
} from "react-router-dom";

import { NavBar } from './NavBar'; //Importamos el NavBar

//Rutas
import { HomeScreen } from './HomeScreen';
import { AboutScreen } from './AboutScreen';
import { LoginScreen } from './LoginScreen';

export const AppRouter = () => {
    return (
        <Router>
            <div>
                <NavBar/>
                <div className="container">
                    <Switch>
                        <Route exact path="/" component={HomeScreen}/>
                        <Route exact path="/about" component={ AboutScreen } />
                        <Route exact path="/login" component={ LoginScreen }/>
                        <Route component={ HomeScreen } />
                    </Switch>
                </div>
            </div>
        </Router>
    )
}
