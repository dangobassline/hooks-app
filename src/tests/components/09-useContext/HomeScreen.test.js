import React from 'react';
import { mount} from 'enzyme';
import { HomeScreen } from '../../../components/09-useContext/HomeScreen';
import { UserContext } from '../../../components/09-useContext/UserContext';
import '@testing-library/jest-dom';

describe('Pruebas en <HomeScreen />', () => {

    const user = {
        name: 'Dango',
        email: 'dango@mail.com'
    }

    const wrapper = mount(
        <UserContext.Provider value={{
                user:user
        }}>
            <HomeScreen />
        </UserContext.Provider>
        );


    test('should de mostrar el componente', () => {
        expect (wrapper).toMatchSnapshot();
    });

})
