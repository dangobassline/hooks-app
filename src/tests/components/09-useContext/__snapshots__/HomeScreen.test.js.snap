// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <HomeScreen /> should de mostrar el componente 1`] = `
<div>
  <h1>
    HomeScreen
  </h1>
  <hr />
  <pre
    className="container"
  >
    {
   "name": "Dango",
   "email": "dango@mail.com"
}
  </pre>
</div>
`;
